extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -386,20 +387,36 @@ void execute(ZonedDateTime now, Vertx vertx) {
                 return;
             }
             ZonedDateTime scheduledFireTime = trigger.evaluate(now);
+            Context context = VertxContext.getOrCreateDuplicatedContext(vertx);
+            VertxContextSafetyToggle.setContextSafe(context, true);
```

Comment:
Why have you moved this code outside the `if (scheduledFireTime != null)` branch? If `scheduledFireTime == null` then the job should not be executed at all and so it does not make sense to call `VertxContextSafetyToggle.setContextSafe()`...